artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -1317,6 +1317,21 @@ public Object getBrokerProperty(SimpleString key) {
       return extra.getProperty(key);
    }
 
+   @Override
+   public final org.apache.activemq.artemis.api.core.Message setIngressTimestamp() {
+      setMessageAnnotation(AMQPMessageSupport.INGRESS_TIMESTAMP, System.currentTimeMillis());
```

Comment:
I see the impl changed to supplying epoch based millisecond value.

If this is going to be the case then I'd say it shouldnt use the same x-opt-ingress-timestamp annotation key as the qpidd broker the feature inspiration came from, since the two would then be using the same key for two differnet styles of value. That would rather break interop between applications and/or the two brokers (at least without significant hoop jumping), rather defeating the point of using the same protocol and annotations.